src/components/shared/ListField.tsx

```diff
@@ -27,21 +28,56 @@ const ListField: FC<IListFieldProps> = ({
 
   function getSapCostValue(field: IForm):string {
     let sapValue = 0;
-
-    if (field.name === 'realizedCost') {
-      sapValue = field.sapCosts?.project_task_costs || 0;
+    if (field.name === 'realizedCostCurrentYear') {
+        const projectTaskCosts = Number(field.sapCurrentYear?.project_task_costs) || 0;
+        const productionTaskCosts = Number(field.sapCurrentYear?.production_task_costs) || 0;
+        sapValue = projectTaskCosts + productionTaskCosts;
+    }
+    else if (field.name === 'realizedCost') {
+      const projectTaskCosts = Number(field.sapCosts?.project_task_costs) || 0;
+      const productionTaskCosts = Number(field.sapCosts?.production_task_costs) || 0;
+      sapValue = projectTaskCosts + productionTaskCosts;
     }
     else if (field.name === 'comittedCost') {
-      sapValue = field.sapCosts?.project_task_commitments || 0;
+      const project_task_commitments = Number(field.sapCosts?.project_task_commitments) || 0;
+      const production_task_commitments = Number(field.sapCosts?.production_task_commitments) || 0;
```

Comment:
The getSapCosts value could be refactored to be more simple by moving the calculation logic to a separate function and using switch-case structure instead of if-else. For example
```
function sumCosts(costs: IProjectSapCost | undefined | null, costType1: string, costType2: string): number {
    return costs ? (Number(costs[costType1 as keyof IProjectSapCost]) || 0) + (Number(costs[costType2 as keyof IProjectSapCost]) || 0) : 0;
  }

  function getSapCostValue(field: IForm): string {
    let sapValue = 0;

    switch (field.name) {
      case 'realizedCostCurrentYear':
        sapValue = sumCosts(field.sapCurrentYear, 'project_task_costs', 'production_task_costs');
        break;

      case 'realizedCost':
        sapValue = sumCosts(field.sapCosts, 'project_task_costs', 'production_task_costs');
        break;

      case 'comittedCost':
        sapValue = sumCosts(field.sapCosts, 'project_task_commitments', 'production_task_commitments');
        break;

      case 'spentCost':
        sapValue = sumCosts(field.sapCosts, 'project_task_costs', 'production_task_costs') + sumCosts(field.sapCosts, 'project_task_commitments', 'production_task_commitments');
        break;

      default:
        break;
    }

    return Number(sapValue).toFixed(0);
  };
```